### PR TITLE
Fixed linters & tests in Mongo, Solr, Badger datasources

### PR DESCRIPTION
**Pull request by coolwednesday**
## Pull Request Template

**Description:**

-   This is a prerequisite to #964
-    We need to fix linters and tests in all datasources before adding workflow for external datasources

**Checklist:**

-   [x] I have formatted my code using  `goimport`  and  `golangci-lint`.
-   [x] All new code is covered by unit tests.
-   [x] This PR does not decrease the overall code coverage.
-   [x] I have reviewed the code comments and documentation for clarity.

**Thank you for your contribution!**

### CODE REVIEW

File: pkg/gofr/datasource/solr/solr.go [hunk 1]
```diff
@@ -153,70 +151,70 @@ func (c *Client) ListFields(ctx context.Context, collection string, params map[s
...
+	resp, span, err := c.call(ctx, http.MethodPost, url, nil, document)
 
-	c.sendOperationStats(ctx, &QueryLog{Type: "DeleteField", Url: url}, startTime, "delete-field", span)
+	c.sendOperationStats(ctx, &QueryLog{Type: "DeleteField", URL: url}, startTime, "delete-field", span)
 
 	return resp, err
 }
 
-// Response stores the response from SOLR
+// Response stores the response from SOLR.
```
Comment (ccoVeille) [inline pkg/gofr/datasource/solr/solr.go:210]:
```suggestion
// Response stores the response from Solr.
```

Solr is not an acronym, but a brand, written Solr

File: pkg/gofr/datasource/solr/solr_test.go [hunk 2]
```diff
@@ -177,9 +177,9 @@ func Test_ClientUpdateField(t *testing.T) {
 func Test_ClientDeleteField(t *testing.T) {
 	s := setupClient(t)
 
-	body := bytes.NewBuffer([]byte(`{"delete-field":{
+	body := bytes.NewBufferString(`{"delete-field":{
```
Comment (ccoVeille) [inline pkg/gofr/datasource/solr/solr_test.go:180]: Did you use mirror linter to detect these?

**Review by ccoVeille — APPROVED** [review]

File: pkg/gofr/datasource/mongo/mongo.go [hunk 3]
```diff
@@ -54,7 +54,7 @@ i.e. by default observability features gets initialised when used with GoFr.
 // client.UseLogger(loggerInstance)
 // client.UseMetrics(metricsInstance)
 // client.Connect().
-func New(c Config) *Client {
+func New(c *Config) *Client {
 	return &Client{config: c}
 }
```
[the remark this answers is not in the document]
Comment (coolwednesday) [reply pkg/gofr/datasource/mongo/mongo.go:59]: the golangci-lint returns : `hugeParam: c is heavy (96 bytes); consider passing it by pointer (gocritic)`
Hence made it a pointer.

File: pkg/gofr/datasource/solr/solr_test.go (hunk 2 above)
Comment (coolwednesday) [reply pkg/gofr/datasource/solr/solr_test.go:180]: Yes, I did check that each such instance is changed according to the linter.

**Review by ccoVeille — APPROVED** [review]

**Review by vikash — APPROVED** [review]

File: pkg/gofr/datasource/solr/solr.go [hunk 4]
```diff
@@ -35,8 +35,8 @@ type Client struct {
 // client := New(config)
 // client.UseLogger(loggerInstance)
 // client.UseMetrics(metricsInstance)
-// client.Connect()
-func New(conf Config) *Client {
+// client.Connect().
+func New(conf *Config) *Client {
```
Comment (vikash) [inline pkg/gofr/datasource/solr/solr.go:39]: This change is unnecessary and does not feel like a good idea. Config should not be a pointer.

Comment (ccoVeille) [reply pkg/gofr/datasource/solr/solr.go:39]: I agree.

https://github.com/gofr-dev/gofr/pull/1245#discussion_r1860112867

Comment (coolwednesday) [reply pkg/gofr/datasource/solr/solr.go:39]: I was trying to be consistent across all the datasources. Some datasources that have a heavy config struct are passed by reference. So was being consistent across datasources passing each config struct by reference.

Comment (ccoVeille) [reply pkg/gofr/datasource/solr/solr.go:39]: The same point has been raised by myself, on another PR

https://github.com/gofr-dev/gofr/pull/1254#discussion_r1862569427

Comment (ccoVeille) [reply pkg/gofr/datasource/solr/solr.go:39]: I don't know what should be done. I agree it's better to have a consistent say to do.

But having a pointer for a configuration looks strange, not, yes it has performance/memory issues

Comment (coolwednesday) [reply pkg/gofr/datasource/solr/solr.go]: Would be changing it to a pointer to struct.

**Review by ccoVeille — APPROVED** [review]

File: pkg/gofr/datasource/mongo/mongo_test.go [hunk 5]
```diff
@@ -536,16 +536,16 @@ func TestClient_StartSession(t *testing.T) {
...
 		err = ses.CommitTransaction(context.Background())
 
-		assert.Nil(t, err)
+		require.NoError(t, err)
 
 		ses.EndSession(context.Background())
 
 		// Assert that there was no error
-		assert.Nil(t, err)
+		require.NoError(t, err)
```
Comment (Umang01-hash) [inline pkg/gofr/datasource/mongo/mongo_test.go:548]: We are using both `assert` as well as `require` in the test file. We can use any one of them throughout.

**Review by Umang01-hash — CHANGES_REQUESTED** [review]

File: pkg/gofr/datasource/mongo/mongo_test.go (hunk 5 above)
Comment (ccoVeille) [reply pkg/gofr/datasource/mongo/mongo_test.go:548]: Do you mean stop using assert?abd only use require package?

It seems a regression. Having tests that fails on first issue is not the best thing

Thi k about checking the http.Status, then the content of the JSON returned

If the check on http.Status fail, you have no idea what the problem is without tjejson content

Comment (coolwednesday) [reply pkg/gofr/datasource/mongo/mongo_test.go:548]: @Umang01-hash thoughts ?

Comment (Umang01-hash) [reply pkg/gofr/datasource/mongo/mongo_test.go:548]: ahhh! Got it 
Sorry my bad! Let's use both

**Review by Umang01-hash — APPROVED** [review]

**Review by vipul-rawat — APPROVED** [review]

**Pull request merged**